server/services/v1/query_runner.go

```
@@ -720,6 +730,100 @@ func (runner *SearchQueryRunner) getFieldSelection(collFields []*schema.Queryabl
+	db, err := runner.GetDatabase(ctx, tx, tenant, runner.req.GetDb())
+	if ulog.E(err) {
+		return nil, ctx, err
+	}
+
+	collection, err := runner.GetCollections(db, runner.req.GetCollection())
+	if ulog.E(err) {
+		return nil, ctx, err
+	}
+
+	table, err := runner.encoder.EncodeTableName(tenant.GetNamespace(), db, collection)
+	if ulog.E(err) {
+		return nil, ctx, err
+	}
+
+	startTime := time.Now().UTC().Format(time.RFC3339Nano)
+	endTime := time.Now().AddDate(34, 0, 0).Format(time.RFC3339Nano)
+	endKey, err := runner.encoder.EncodeKey(table, collection.Indexes.PrimaryKey, []interface{}{endTime})
+	if ulog.E(err) {
+		return nil, ctx, err
+	}
+
+	ticker := time.NewTicker(500 * time.Millisecond)
+	defer ticker.Stop()
+
+	for range ticker.C {
+		tickerTx, err := runner.txMgr.StartTx(ctx)
+		if ulog.E(err) {
+			return nil, ctx, err
+		}
+
+		startKey, err := runner.encoder.EncodeKey(table, collection.Indexes.PrimaryKey, []interface{}{startTime})
+		if ulog.E(err) {
+			return nil, ctx, err
+		}
+
+		kvIterator, err := tickerTx.ReadRange(ctx, startKey, endKey)
+		if ulog.E(err) {
+			return nil, ctx, err
+		}
+
+		var last []byte
+		for {
+			var keyValue kv.KeyValue
+			if kvIterator.Next(&keyValue) {
+				err = runner.streaming.Send(&api.SubscribeResponse{
+					Message: keyValue.Data.RawData,
+				})
+				if ulog.E(err) {
+					return nil, ctx, err
+				}
+
+				last = keyValue.Data.RawData
+			} else {
+				break
+			}
+
+			if ulog.E(kvIterator.Err()) {
+				return nil, ctx, kvIterator.Err()
+			}
+		}
+
+		if last != nil {
+			var data map[string]any
+			err = jsoniter.Unmarshal(last, &data)
+			if ulog.E(err) {
+				return nil, ctx, err
+			}
+
+			lastTime, err := time.Parse(time.RFC3339, data[schema.AutoPrimaryKeyF].(string))
+			if ulog.E(err) {
+				return nil, ctx, err
+			}
+
+			startTime = lastTime.Add(1 * time.Nanosecond).Format(time.RFC3339)
+		}
+
+		err = tickerTx.Commit(ctx)
```

add a ToDo that as this is a read-only transaction we should ignore any error like transaction timed-out etc.